### PR TITLE
feat(CI): update CI to push images to quay.io instead of dockerhub

### PR DESCRIPTION
**Pull request by damienomurchu**
## Motivation
Host images on quay.io instead of dockerhub

## What
Host images on quay.io instead of dockerhub

## Why
Aerogear images to be hosted on quay.io

## How
Update push command in makefile & add quay.io credentials to project in CircleCI dashboard

## Verification Steps
- deploy to your local environment
```
minishift start
oc login -u system:admin
make create-all
make create-app
``` 
- update MSS image to point to testing image: `quay.io/damienomurchu/mobile-security-service:c29f31e`
- check that pod redeploys correctly
- check that no functionality has been broken (JIRA [here](https://issues.jboss.org/browse/AEROGEAR-9233) is a good reference as to what to check)
- check that the UI renders the app(s) in the table and the logged in user in the header correctly

## Checklist:

- [x] Code has been tested locally by PR requester
- [ ] Changes have been successfully verified by another team member 

## Progress

- [x] Finished task

### CODE REVIEW

**Comment by damienomurchu** [comment]
Thanks for the quick approval @camilamacedo86 - PR to update operator is here https://github.com/aerogear/mobile-security-service-operator/pull/113 if you can review too

**Comment by damienomurchu** [comment]
Just running some small checks before I merge, fyi

File: Makefile [hunk 1]
```diff
@@ -65,13 +66,13 @@ docker_build_master:
 
 .PHONY: docker_push_release
 docker_push_release:
-	@docker login --username $(DOCKERHUB_USERNAME) --password $(DOCKERHUB_PASSWORD)
+	@docker login --username $(QUAY_USERNAME) --password $(QUAY_PASSWORD) quay.io
```
Comment (laurafitzgerald) [inline Makefile]: We could use IMAGE_REGISTRY variable here instea dof hardcoding `quay.io`

File: Makefile [hunk 2]
```diff
@@ -65,13 +66,13 @@ docker_build_master:
...
 	docker push $(DOCKER_LATEST_TAG)
 	docker push $(DOCKER_RELEASE_TAG)
 
 .PHONY: docker_push_master
 docker_push_master:
-	@docker login --username $(DOCKERHUB_USERNAME) --password $(DOCKERHUB_PASSWORD)
+	@docker login --username $(QUAY_USERNAME) --password $(QUAY_PASSWORD) quay.io
```
Comment (laurafitzgerald) [inline Makefile]: same here

File: Makefile [hunk 3]
```diff
@@ -65,13 +66,13 @@ docker_build_master:
 
 .PHONY: docker_push_release
 docker_push_release:
-	@docker login --username $(DOCKERHUB_USERNAME) --password $(DOCKERHUB_PASSWORD)
+	@docker login --username $(QUAY_USERNAME) --password $(QUAY_PASSWORD) quay.io
 	docker push $(DOCKER_LATEST_TAG)
```
Comment (laurafitzgerald) [inline Makefile]: Why do we push latest and release tag both here for `docker_push_release`?

File: Makefile [hunk 4]
```diff
@@ -65,13 +66,13 @@ docker_build_master:
 
 .PHONY: docker_push_release
```
Comment (laurafitzgerald) [inline Makefile]: can we change the name of these commands?.

File: Dockerfile [hunk 5]
```diff
@@ -1,4 +1,4 @@
-FROM openshift/origin-release:golang-1.10 as builder
+FROM openshift/origin-release:golang-1.12 as builder
```
Comment (laurafitzgerald) [inline Dockerfile]: What level of testing do we need to do in order to ensure a change to the go version is okay?

File: Makefile (hunk 3 above)
Comment (damienomurchu) [reply Makefile]: This is to publish an image with the `<release-tag>`, and also to publish an image with the `latest` tag, ie so you can either pull a specific version, or just the latest tagged version

File: Makefile (hunk 4 above)
Comment (damienomurchu) [reply Makefile]: Its possible. I would not see `docker_push_release` referring to the registry but rather the way in which we push the image
/2cents

File: Dockerfile (hunk 5 above)
Comment (damienomurchu) [reply Dockerfile]: Good question. For this PR I will verify all functionality of the MSS after updating to this image so as to ensure no broken functionality.

**Comment by damienomurchu** [comment]
@laurafitzgerald think I've addressed all of your requested changes and have also checked that no functionality has been broken by the update to use the images from `quay.io` or the update to use a later base image for the mss image

If you / @camilamacedo86 could re-review that would be awesome :)

**Comment by damienomurchu** [comment]
Thanks @davidffrench @camilamacedo86 @laurafitzgerald - just need an approval now if you don't mind doing the honours (to confirm: I won't be merging this until later, so you can feel free to approve now)

**Pull request merged**